sound/soc/sof/ipc4.c

```diff
@@ -219,6 +219,15 @@ int sof_ipc4_init_msg_memory(struct snd_sof_dev *sdev)
 
 	/* TODO: get max_payload_size from firmware */
 	sdev->ipc->max_payload_size = SOF_IPC4_MSG_MAX_SIZE;
+	/*
+	 * In kernel a full IPC4 message is calculated to take the
+	 * struct sof_ipc4_msg size:
+	 * msg_total_size = sizeof(msg) + msg.data_size;
+	 *
+	 * The msg_total_size should be passed as msg_bytes for function expecting
+	 * a message size and underlaying code should take this into account.
+	 */
+	sdev->ipc->max_payload_size += sizeof(struct sof_ipc4_msg);
```

Comment:
I have missed the setup of the reply placeholder, this should be added after line 227:
```c
	ipc4_msg = msg->reply_data;
	ipc4_msg->data_ptr = msg->reply_data + sizeof(struct sof_ipc4_msg);
```
So that the data_ptr points to the memory right after the header.

Or I could have two separate allocations for the message struct then the data section?